Add cover component support to emulated hue

roidayan opened the pull request:
Support for turn on/off which will open/close the cover component.
Also add cover to default exposed list.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

**Pull request in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io) with documentation:** home-assistant/home-assistant.github.io#1646

**Checklist:**

  - [x] Documentation added/updated in [home-assistant.github.io](https://github.com/home-assistant/home-assistant.github.io)

roidayan committed:
Add cover component support to emulated hue

Support for turn on/off which will open/close the cover component.
Also add cover to default exposed list.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

trisk reviewed (commented):
Could also move the first `hass.async_add_job` call to the media_player section since that's the only consumer right now.

roidayan commented:
hi @trisk, can you tell me why it exists? is the point of it is to turn on the media player if we actually wanted to change volume? is so is this even a desired action?

trisk commented:
The only reason it exists is for consistency with the semantics of lights, where if you dim the lights they also turn on first. Since we can't do a delay here it doesn't work with all devices anyway, so I'd be open to just removing it. The test case for media_player would need to be altered slightly.

roidayan committed:
Do not auto turn on a media player when requesting to set the volume

A user should first ask to turn the media player on.
The behavior of doing multiple actions like in the case for turning on
the media player and set the volume can be done using a script.

Updated the test to match this change.

Signed-off-by: Roi Dayan <roi.dayan@gmail.com>

roidayan commented:
hi @trisk , I updated the test and it's passing.

trisk commented:
Yep that seems fine. Thanks.

balloob commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -205,12 +212,6 @@ def put(self, request, username, entity_number):
             # as the actual requested command.
             config.cached_states[entity_id] = (result, brightness)
 
-        # Separate call to turn on needed
```
Why did you remove this?

trisk replied:
That was my suggestion since the separate on/brightness was not terribly useful for real devices. It also alters the semantics for the test.

balloob commented on homeassistant/components/emulated_hue/hue_api.py:
```diff
@@ -193,6 +194,12 @@ def put(self, request, username, entity_number):
                     service = SERVICE_VOLUME_SET
                     # Convert 0-100 to 0.0-1.0
                     data[ATTR_MEDIA_VOLUME_LEVEL] = brightness / 100.0
+        elif entity.domain == "cover":
```
Can you add a test that changes the state of a cover?

roidayan replied:
ok will do

balloob commented on tests/components/emulated_hue/test_hue_api.py:
```diff
@@ -240,10 +240,10 @@ def test_put_light_state_script(self):
     def test_put_light_state_media_player(self):
-        """Test turning on media player and setting volume."""
-        # Turn the music player off first
+        """Test setting media player volume."""
```
What's with these changes?

roidayan replied:
this is updated doc to the change trisk suggested. so the test is not turning off the media player and testing that changing the volume will auto turn it on as it will not.

roidayan commented:
redundant now. i see another pull request like this was merged
